modules/Bio/EnsEMBL/Variation/DBSQL/PublicationAdaptor.pm

```diff
@@ -348,9 +352,9 @@ sub update_variant_citation {
         ## avoid duplicates
         $citation_ext_sth->execute( $var_obj->dbID(),  $pub->{dbID} );
         my $count = $citation_ext_sth->fetchall_arrayref();
-        next unless $count->[0]->[0] ==0;
-
-        $citation_ins_sth->execute( $var_obj->dbID(), $pub->{dbID});
+        next unless $count->[0]->[0] ==0;  
+        $citation_ins_sth->execute( $var_obj->dbID(), $pub->{dbID} );
```

Comment:
Not clear why there is an insert  followed by an update - could it not be done as one step?

Reply:
Looks better. Thanks!